scripts/pre_commit_linter.py

```diff
@@ -1197,6 +1198,54 @@ def _check_directive_scope(self):
 
         return summary_messages
 
+    def _check_js_file_overview(self):
```

Comment:
Can we merge this with an existing check (maybe the copyright notice check)?

Reply:
I am not quite sure about that, because both the functions have a lot of uncommon logic, and since this function is quite big too, I dont think merging them is a good idea. Also having different functions for different purposes is a good idea in my opinion.
What do you think?

Reply:
Okay, I think we can simply add a regex for this check to `BAD_PATTERNS_JS_REGEXP` and do away with this function completely. Let me know your opinions. Thanks!

Reply:
The `BAD_PATTERNS_JS_REGEXP` is something which is not allowed, whereas the file overviews are something which are to be checked for its presence. Also, the logic that uses `BAD_PATTERNS_JS_REGEXP` checks for the pattern in every line of the file which is not the case we want here. The workaround would be to have a condition which would check against this particular regex and stop further checking of the file for this very regex once a match is found. But I think this would destroy the logic of having `BAD_PATTERNS_JS_REGEXP` in the first place. Also, by using the other function, we can directly extract the comments (and its type i.e. 'block' or 'line') due to esprima. 
What do you think about this?

Reply:
Hi @apb7, Any thoughts on this?